terraform/main.tf

```diff
@@ -79,70 +79,149 @@ locals {
   additional_disks = concat(local.fs_disks, local.asm_disks)
 
   project_id = var.project_id
+
+  is_multi_instance = (
+    var.zone1 != "" && var.zone2 != "" && var.subnetwork1 != "" && var.subnetwork2 != ""
+  )
+
+  instances = local.is_multi_instance ? {
+    node1 = {
+      name       = "${var.instance_name}-1"
+      zone       = var.zone1
+      subnetwork = var.subnetwork1
+    }
+    node2 = {
+      name       = "${var.instance_name}-2"
+      zone       = var.zone2
+      subnetwork = var.subnetwork2
+    }
+  } : {
+    default = {
+      name       = var.instance_name
```

Comment:
Just had a thought here:  in the future, might we run into a situation where someone wants to add a standby to an existing single-instance database?  Would we change the hostname of the existing primary at that point?  Or just leave it at ${var.instance_name} and a new ${var.instance_name}-2?  The alternative would be to name even single-node as ${var.instance_name}-1.